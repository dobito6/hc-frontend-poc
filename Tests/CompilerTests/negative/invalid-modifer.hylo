//! no-std

static trait P {

  public type X

}

static struct A {

  static typealias Z = A

}

public extension A {

  public static public fun f0() {}

  static public fun f0() {}

}
