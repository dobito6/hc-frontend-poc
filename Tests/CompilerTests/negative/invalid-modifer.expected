negative/invalid-modifer.hylo:1.1-7: error: declaration cannot be marked 'static'
static trait P {
~~~~~~
negative/invalid-modifer.hylo:3.3-9: error: declaration cannot be marked 'public'
  public type X
  ~~~~~~
negative/invalid-modifer.hylo:7.1-7: error: declaration cannot be marked 'static'
static struct A {
~~~~~~
negative/invalid-modifer.hylo:9.3-9: error: declaration cannot be marked 'static'
  static typealias Z = A
  ~~~~~~
negative/invalid-modifer.hylo:13.1-7: error: declaration cannot be marked 'public'
public extension A {
~~~~~~
negative/invalid-modifer.hylo:15.17-23: error: duplicate modifier
  public static public fun f0() {}
                ~~~~~~
negative/invalid-modifer.hylo:17.10-16: error: 'public' should occur before 'static'
  static public fun f0() {}
         ~~~~~~
